src/js/controller/controller.js

```diff
@@ -1036,15 +1044,14 @@ Object.assign(Controller.prototype, {
         };
 
         this.setPlaylistItem = function (index, item) {
-            item = normalizePlaylistItem(_model, new Item(item), item.feedData || {});
-
-            if (item) {
-                const playlist = _model.get('playlist');
-                playlist[index] = item;
-
-                if (index === _model.get('item') && _model.get('state') === 'idle') {
-                    this.setItemIndex(index);
+            const newItem = _programController.setPlaylistItem(index, item);
+            if (newItem && index === _model.get('item') && _model.get('state') === 'idle') {
+                const itemPromise = this.itemPromises[index];
```

Comment:
it looks like `itemPromises` only exists on `programController` - is there some magic i'm missing that's making `this` share context here? otherwise i don't see how this would get populated

Reply:
No magic. Good catch. Fixed in 46c3e77 ...and the commit after that.

If the user calls `jwplayer().setPlaylistItem(currentIndex, { file: 'video' })` while the player is idle, we will reload the current item. We should always clear the item promise when items are replaced in the api>controller.